### PR TITLE
Adding Navigation & BookFilter components

### PR DESCRIPTION
**Pull request by SiantyTown**
an attempt was made~

### CODE REVIEW

File: FrontEnd/src/components/CheckBoxCustom.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ class CheckBoxCustom extends React.Component {
   render() {
     return (
       
-      <div class="check-box">
+      <div className="check-box">
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/CheckBoxCustom.js:13]: did you checked if this is not clashing with the framework checkbox?

File: FrontEnd/src/components/CheckBoxCustom.js [hunk 2]
```diff
@@ -24,13 +24,22 @@ class CheckBoxCustom extends React.Component {
...
   }
 
+  updateState () {
+    this.setState({
+        elements: Object.assign({}, this.state.elements[1], {
+            disabled: 'true'
+        })
+    });
+  }
+
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/CheckBoxCustom.js]: not quite sure why you need this / what this is doing
you're never calling it?

File: FrontEnd/src/components/Navigation.js [hunk 3]
```diff
@@ -8,71 +8,75 @@ import { Row, Col } from 'react-materialize';
...
-			case 'Settings':
-				this.state.pages[1].class += ' active';
-				break;
+    render() {
+        return (
+            <div className={'navigation'}>
+                <Row>
+                    <Col s={12}>
+                        <ul className="mt0 mb0">
+                            {this.state.pages.map((navList) => (
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/Navigation.js]: minor thing, but this is not really a 'list' but an 'object' or 'element'

File: FrontEnd/src/components/Navigation.js [hunk 4]
```diff
@@ -8,71 +8,75 @@ import { Row, Col } from 'react-materialize';
...
+                                </li>
+                            ))}
+                        </ul>
+                    </Col>
+                </Row>
+            </div>
+        );
+    }
+    
+    switchSelected(id) {
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/Navigation.js:65]: you need to bind all "internal functions"

File: FrontEnd/src/components/Navigation.js [hunk 5]
```diff
@@ -8,71 +8,75 @@ import { Row, Col } from 'react-materialize';
...
+        this.setState(prevState => {
+            selectedNew = this.state.pages.find(e => e.text === id);
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/Navigation.js]: shouldn't this be a `:` ? O.o
also, all `this.state` inside this function should really be `prevState`

File: FrontEnd/src/components/CheckBoxCustom.js (hunk 1 above)
Comment (SiantyTown) [reply FrontEnd/src/components/CheckBoxCustom.js:13]: it doesn't appear to be? I don't think they put a dash in it but i'm not sure

File: FrontEnd/src/components/CheckBoxCustom.js (hunk 2 above)
Comment (SiantyTown) [reply FrontEnd/src/components/CheckBoxCustom.js]: forgot to remove it

File: FrontEnd/src/components/BookFilter.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+import componentCSS from './BookFilter.scss';
+
+class BookFilter extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			bookStatusFilters: [
+				{
+					id: 'requested',
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/BookFilter.js]: there's an enum for these =$

File: FrontEnd/src/components/BookFilter.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+				</Row>
+				<Row>
+					<Col s={12}>
+						<div className="filters">
+							<span>Status</span>
+
+							{this.state.bookStatusFilters.map((e) => (
+								<CheckBoxCustom
+									key={uuidv4()}
+									filledIn
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/BookFilter.js]: don't think you need this here

File: FrontEnd/src/components/BookFilter.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+						</div>
+					</Col>
+				</Row>
+			</div>
+		);
+	}
+
+	updateState(id) {
+		this.setState((prevState) => {
+			let selected = this.state.bookStatusFilters.find((e) => e.id === id);
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/BookFilter.js]: `this.state` should be `prevState`

File: FrontEnd/src/components/Navigation.js (hunk 3 above)
Comment (SiantyTown) [reply FrontEnd/src/components/Navigation.js]: done this way as that's how I've typically seen navigation done but i can update it to a div?

File: FrontEnd/src/components/BookFilter.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+								/>
+							))}
+						</div>
+					</Col>
+				</Row>
+			</div>
+		);
+	}
+
+	updateState(id) {
```
Comment (ZoltanT-RD) [inline FrontEnd/src/components/BookFilter.js:98]: internal function needs binding

File: FrontEnd/src/components/Navigation.js (hunk 4 above)
Comment (SiantyTown) [reply FrontEnd/src/components/Navigation.js:65]: think this is done

File: FrontEnd/src/components/Navigation.js (hunk 5 above)
Comment (SiantyTown) [reply FrontEnd/src/components/Navigation.js]: I have done half of this [commenting because i keep forgetting which i have done]

**Review by ZoltanT-RD — APPROVED** [review]

**Pull request merged**